WalletWasabi.Gui/Tabs/WalletManager/GenerateWalletViewModel.cs

```diff
@@ -47,61 +44,23 @@ public GenerateWalletViewModel(WalletManagerViewModel owner) : base("Generate Wa
 
 		private void DoNextCommand()
 		{
-			WalletName = Guard.Correct(WalletName);
-
-			if (!ValidateWalletName(WalletName))
-			{
-				NotificationHelpers.Error("Invalid wallet name.");
-				return;
-			}
-
-			string walletFilePath = Path.Combine(Global.WalletsDir, $"{WalletName}.json");
-
-			if (!TermsAccepted)
-			{
-				NotificationHelpers.Error("Terms are not accepted.");
-			}
-			else if (string.IsNullOrWhiteSpace(WalletName))
-			{
-				NotificationHelpers.Error("Invalid wallet name.");
-			}
-			else if (File.Exists(walletFilePath))
-			{
-				NotificationHelpers.Error("Wallet name is already taken.");
-			}
-			else
+			try
 			{
-				try
+				if (!TermsAccepted)
 				{
-					PasswordHelper.Guard(Password); // Here we are not letting anything that will be autocorrected later. We need to generate the wallet exactly with the entered password because of compatibility.
-
-					var km = KeyManager.CreateNew(out Mnemonic mnemonic, Password);
-					km.SetNetwork(Global.Network);
-					km.SetBestHeight(new Height(Global.BitcoinStore.SmartHeaderChain.TipHeight));
-					km.SetFilePath(walletFilePath);
-					Owner.CurrentView = new GenerateWalletSuccessViewModel(Owner, km, mnemonic);
-				}
-				catch (Exception ex)
-				{
-					Logger.LogError(ex);
-					NotificationHelpers.Error(ex.ToUserFriendlyString());
+					throw new InvalidOperationException("Terms are not accepted.");
```

Comment:
Is it better to `throw new ..` or to use `NotificationHelpers.Error`?

Reply:
It is caught later and there is the notification.